version.go

```diff
@@ -0,0 +1,33 @@
+package main
+
+import (
+	_ "embed"
+	"encoding/json"
+	"runtime/debug"
+)
+
+var version string
+
+//go:embed version.json
+var versionJSON []byte
+
+func init() {
+	// Read version from embedded JSON file.
+	var verMap map[string]string
+	json.Unmarshal(versionJSON, &verMap)
+	version = verMap["version"]
+
+	// If running from a module, try to get the build info.
+	bi, ok := debug.ReadBuildInfo()
+	if !ok {
+		return
+	}
+
+	// Append the revision to the version.
+	for i := range bi.Settings {
+		if bi.Settings[i].Key == "vcs.revision" {
```

Comment:
Should probably also include a dirty flag like in these:

https://github.com/ipni/index-provider/blob/c282b0aebc35233153863e3d260d40184f26c2bf/version.go#L44

https://github.com/ipfs/rainbow/blob/471f09587d51044070a12bfc2df7c3092fc9c581/version.go#L39

Reply:
Done - Using logic from rainbow's version.go. I think this should be the standard elsewhere.